pkg/sharky/slots.go

```diff
@@ -10,38 +10,33 @@ import (
 )
 
 type slots struct {
-	data    []byte          // byteslice serving as bitvector: i-t bit set <>
-	size    uint32          // number of slots
-	head    uint32          // the first free slot
-	file    sharkyFile      // file to persist free slots across sessions
-	in      chan uint32     // incoming channel for free slots,
-	out     chan uint32     // outgoing channel for free slots
-	wg      *sync.WaitGroup // count started write operations
-	limboWG sync.WaitGroup  // wait for the limbo writes to in chan after the quit is closed
+	data []byte     // byteslice serving as bitvector: i-t bit set <>
+	head uint32     // the first free slot
+	file sharkyFile // file to persist free slots across sessions
+	mtx  sync.Mutex
 }
 
-func newSlots(file sharkyFile, wg *sync.WaitGroup) *slots {
+func newSlots(file sharkyFile) *slots {
 	return &slots{
 		file: file,
-		in:   make(chan uint32),
-		out:  make(chan uint32),
-		wg:   wg,
 	}
 }
 
-// load inits the slots from file, called after init
-func (sl *slots) load() (err error) {
-	sl.data, err = io.ReadAll(sl.file)
+func (sl *slots) load() error {
+	data, err := io.ReadAll(sl.file)
 	if err != nil {
 		return err
 	}
-	sl.size = uint32(len(sl.data) * 8)
-	sl.head = sl.next(0)
-	return err
+	sl.data = data
```

Comment:
Should statement 

```sl.data = data```

be guarded with mutex?

Reply:
no, this function is only used during bootup so no mutex is needed